src/courtroom.cpp

```diff
@@ -1788,89 +1813,51 @@ void Courtroom::handle_chatmessage(QStringList *p_contents)
   }
 
   int f_char_id = m_chatmessage[CHAR_ID].toInt();
+  const bool is_spectator = (f_char_id == -1);
 
-  if (f_char_id >= 0 && f_char_id >= char_list.size())
+  if (f_char_id >= char_list.size())
```

Comment:
what if it's -3

Reply:
-3 will never be >= char_list.size() so it doesn't matter

Reply:
fixed to include negative check, which didn't work in the first place